engine/apps/api/views/user.py

```diff
@@ -185,9 +180,12 @@ def get_serializer_class(self):
         is_users_own_data = (
             self.kwargs.get("pk") is not None and self.kwargs.get("pk") == self.request.user.public_primary_key
         )
-        if is_users_own_data or self.request.user.role == Role.ADMIN:
-            return UserSerializer
-        return UserHiddenFieldsSerializer
+
+        # TODO: how to remove role here?
+        # if is_users_own_data or self.request.user.role == Role.ADMIN:
+        #     return UserSerializer
+        # return UserHiddenFieldsSerializer
+        return UserSerializer
```

Comment:
I could use some help refactoring `get_serializer_class` here as the query is currently dependent on `user.role`